awsbestpractices/loops-to-batches/LoopsToBatches.java

```
@@ -0,0 +1,185 @@
+        ItemConstraintViolationException itemConstraintViolationException = null;
+        for (final UniqueConstraint uniqueConstraint : itemConfiguration.uniqueConstraints()) {
+            final String uniqueConstraintPropertyName = uniqueConstraint.propertyName();
+            final PropertyDescriptor uniqueConstraintPropertyDescriptor = uniqueConstraint.propertyDescriptor();
+            if (constraintPropertyDescriptors.contains(uniqueConstraintPropertyDescriptor)) {
+                final AttributeValue uniqueConstraintAttributeValue = DynamoDbPropertyMarshaller.getValue(item,
+                        uniqueConstraintPropertyDescriptor);
+                if (uniqueConstraintAttributeValue == null) {
+                    continue;
+                }
+                if (uniqueConstraintAttributeValue.getS() != null) {
+                    uniqueConstraintAttributeValue.setS(uniqueConstraintAttributeValue.getS().toUpperCase());
+                }
+                final Map<String, AttributeValue> attributeMap = new HashMap<>();
+                attributeMap.put("property", new AttributeValue(uniqueConstraintPropertyName));
+                attributeMap.put("value", uniqueConstraintAttributeValue);
+                final Map<String, ExpectedAttributeValue> expectedResults = new HashMap<>();
+                expectedResults.put("value", new ExpectedAttributeValue(false));
+                final String indexTableName = databaseSchemaHolder.schemaName() + "-indexes."
+                        + itemConfiguration.tableName();
+                final PutItemRequest itemRequest = new PutItemRequest().withTableName(indexTableName)
+                        .withItem(attributeMap).withExpected(expectedResults);
+                try {
+                    dynamoDBClient.putItem(itemRequest);
+                    createdConstraintPropertyDescriptors.add(uniqueConstraintPropertyDescriptor);
+                } catch (final ConditionalCheckFailedException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)